corona-warn-app/cwa-app-android
# Most relevant DCC, update vaccination priority (EXPOSUREAPP-9443)

It's now a feature, not a bug 😄 
The logic for vaccinations has been updated to accomodate boosters.

What has changed from the previous logic: 
1. A BioNTech/Moderna/AstraZeneca vaccine that is taken after a recovery (that means it will have dose 1/1), will now get priority right away. If it's a J&J vaccine, it will get priority after 14 days.
2. A BioNTech/Moderna/AstraZeneca vaccine with dose 3/3 will get priority right away.
3. a J&J vaccine with dose 2/2 will get priority right away.
3. All other cases are covered by the old logic.

How to test:
Start from a clean slate. Delete all other certificates if you have any. Switch to WRU env. I'm using BioNTech and J&J vaccines to test, but you can substitute BioNTech with Moderna or AstaZeneca. 
Moderna: `EU/1/20/1507`
AstraZeneca: `EU/1/21/1529`

1. Scan a recovery certificate < 180 days old: 
 `cwa rc gen --pii 42 -s r.0.fr='2021-08-01' r.0.df='2021-10-10' -e wru`. This will have highest prio as it's the only one.
2. Scan a J&J vaccine < 14 days old:
`cwa vc gen --pii 42 -s v.0.dn=1 v.0.sd=1 v.0.mp='EU/1/20/1525' v.0.dt='2021-11-04' -e wru`. The recovery certificate should still have highest prio as this vaccine is < 14 days old. Vaccination Status info box should say you have received all planned vaccinations but protection won't be complete for x days.
3. Scan a J&J vaccine > 14 days old:
`cwa vc gen --pii 42 -s v.0.dn=1 v.0.sd=1 v.0.mp='EU/1/20/1525' v.0.dt='2021-10-04' -e wru`. This vaccine will now get the highest prio as the 14 days have passed. Vaccination Status info box should say you have received all planned vaccinations and protection is complete.
4. Remove the two J&J vaccines and leave the recovery certificate in place.
5. Scan a BioNTech vaccine with dose 1/1 that's < 14 days old:
`cwa vc gen --pii 42 -s v.0.dn=1 v.0.sd=1 v.0.mp='EU/1/20/1528' v.0.dt='2021-11-04' -e wru`. This vaccine should get highest priority right away. Vaccination Status info box should say you have received all planned vaccinations and protection is complete.
6. Scan a BioNTech vaccine with dose 1/1 that's > 14 days old:
`cwa vc gen --pii 42 -s v.0.dn=1 v.0.sd=1 v.0.mp='EU/1/20/1528' v.0.dt='2021-10-04' -e wru`. The most recent BioNTech vaccine (the one < 14 days from the previous point) should still have highest prio. Vaccination Status info box should say you have received all planned vaccinations and protection is complete.
7. Remove the two BioNTech vaccines.
8. Scan a J&J vaccine > 14 days old:
`cwa vc gen --pii 42 -s v.0.dn=1 v.0.sd=1 v.0.mp='EU/1/20/1525' v.0.dt='2021-10-04' -e wru`. This will have the highest prio (already covered at point 3).
9. Scan a BioNTech vaccine with dose 2/2 that's < 14 days old:
`cwa vc gen --pii 42 -s v.0.dn=2 v.0.sd=2 v.0.mp='EU/1/20/1528' v.0.dt='2021-11-04' -e wru`. The J&J vaccine still has the highest priority as this vaccine is < 14 days old. The Vaccination Status info box should say you have received all planned vaccinations.
10. Scan a BioNTech vaccine with dose 2/2 that's > 14 days old:
`cwa vc gen --pii 42 -s v.0.dn=2 v.0.sd=2 v.0.mp='EU/1/20/1528' v.0.dt='2021-10-04' -e wru`. This vaccine now gets highest priority. The Vaccination Status info box should say you have received all planned vaccinations and protection is complete.
11. Delete all vaccination certificates and leave the recovery certificate in place.
12. Scan a BioNTech certificate with dose 1/1 that's > 14 days old:
`cwa vc gen --pii 42 -s v.0.dn=1 v.0.sd=1 v.0.mp='EU/1/20/1528' v.0.dt='2021-10-04' -e wru`. This vaccine will have highest priority. The Vaccination Status info box should say you have recieved all planned vaccinations and protection is complete.
13. Scan a BioNTech certificate with dose 2/2 that's < 14 days old:
`cwa vc gen --pii 42 -s v.0.dn=2 v.0.sd=2 v.0.mp='EU/1/20/1528' v.0.dt='2021-11-04' -e wru`. The previous BioNTech vaccine with dose 1/1 will still have highest priority as this one is < 14 days old. The Vaccination Status info box should say you have received all planned vaccinations.
14. Scan a BioNTech certificate with dose 2/2 that's > 14 days old:
`cwa vc gen --pii 42 -s v.0.dn=2 v.0.sd=2 v.0.mp='EU/1/20/1528' v.0.dt='2021-10-04' -e wru`. This vaccine will now get highest priority. The Vaccination Status info box should say you have received all planned vaccinations and protection is complete.
15. Remove all certificates.
16. Scan a BioNTech certificate with dose 1/2 that's < 14 days old:
`cwa vc gen --pii 42 -s v.0.dn=1 v.0.sd=2 v.0.mp='EU/1/20/1528' v.0.dt='2021-11-04' -e wru`. This vaccine will have highest prio as it's the only certificate scanned. The Vaccination Status info box should say you have not yet received all planned vaccinations. Your protection is not complete yet.
17. Scan a BioNTech certificate with dose 1/2 that's > 14 days old:
`cwa vc gen --pii 42 -s v.0.dn=1 v.0.sd=2 v.0.mp='EU/1/20/1528' v.0.dt='2021-10-04' -e wru`. The previous vaccine that's < 14 days old will still have highest priority because for incomplete doses we take the priority depending on the most recent date. The Vaccination Status info box should say you have not yet received all planned vaccinations. Your protection is not complete yet.
18. Scan a BioNTech certificate with dose 2/2 that's < 14 days old:
`cwa vc gen --pii 42 -s v.0.dn=2 v.0.sd=2 v.0.mp='EU/1/20/1528' v.0.dt='2021-11-04' -e wru`.  This vaccination will get highest priority as it's a complete vaccination. The Vaccination Status info box should say you have received all planned vaccinations but protection won't be complete for x days.
19. Remove the previous vaccine with dose 2/2 and < 14 days old.
20. Scan a BioNTech certificate with dose 2/2 that's > 14 days old:
`cwa vc gen --pii 42 -s v.0.dn=2 v.0.sd=2 v.0.mp='EU/1/20/1528' v.0.dt='2021-10-04' -e wru`. This vaccine will get highest priority. The Vaccination Status info box should say you have recieved all planned vaccinations and protection is complete.
21. Scan a BioNTech certificate with dose 3/3 that's < 14 days old:
`cwa vc gen --pii 42 -s v.0.dn=3 v.0.sd=3 v.0.mp='EU/1/20/1528' v.0.dt='2021-11-04' -e wru`. This vaccine will get highest priority right away. The Vaccination Status info box should say you have received all planned vaccinations and protection is complete.
22. Remove all certificates.
23. Scan a J&J with dose 1/1 that's > 14 days old:
`cwa vc gen --pii 42 -s v.0.dn=1 v.0.sd=1 v.0.mp='EU/1/20/1525' v.0.dt='2021-10-04' -e wru`. This vaccine will get highest priority as it's the only certificate. The Vaccination Status info box should say you have received all planned vaccinations and protection is complete.
24. Scan a J&J with dose 2/2 that's < 14 days old:
`cwa vc gen --pii 42 -s v.0.dn=2 v.0.sd=2 v.0.mp='EU/1/20/1525' v.0.dt='2021-11-04' -e wru`. This vaccine will get highest priority right away as it's considered a booster. The Vaccination Status info box should say you have received all planned vaccinations.
